src/boot/efi/stub.c

```diff
@@ -249,7 +250,7 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
                 }
         }
 
-        err = linux_exec(image, cmdline, cmdline_len, linux_base, initrd_base, initrd_size);
+        err = linux_exec(image, cmdline, cmdline_len, (VOID*) linux_base, linux_size, (VOID*) initrd_base, initrd_size);
```

Comment:
I changed the signature of linux_exec to use pointers instead of PHYSICAL_ADDRESS because I wanted to avoid conversions. It would be for the best to either change stub.c to also just use pointers. (is this PR the right place for such a change?)

Reply:
doing the conversion in the same PR sounds fine, but please in a separate commit.

Reply:
I'll spit the commits